spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-config/src/main/java/com/alibaba/cloud/nacos/configdata/NacosBootstrapper.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2013-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.cloud.nacos.configdata;
+
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import com.alibaba.nacos.api.config.ConfigService;
+import org.springframework.boot.BootstrapContext;
+import org.springframework.boot.BootstrapRegistry;
+import org.springframework.boot.BootstrapRegistry.InstanceSupplier;
+import org.springframework.boot.BootstrapRegistryInitializer;
+import org.springframework.boot.context.config.ConfigData;
+import org.springframework.boot.context.config.ConfigDataLoaderContext;
+import org.springframework.boot.context.properties.bind.Binder;
+import org.springframework.util.Assert;
+
+/**
+ * Spring boot >= 2.4.5 supports {@link BootstrapRegistryInitializer}.
```

Comment:
when the user boot version is relatively low, if they directly upgrade to your current version, will it be compatible?
e.g:boot version is 2.1.x or 2.2.x , the nacos-config  is latest version

Reply:
This question is a bit confusing, if user uses boot version is 2.1.x or 2.2.x, he should not use SCA 2021 version.

Do you mean if boot version < 2.4.5 is compatible?

If it's this question, then the answer is compatible. This is an extension class, just do not have custom extensions when the version is less than 2.4.5.

Reply:
all current features are based on 2.3.x ;if the SCA version is upgraded to the latest , the feature cannot be used and cannot even be started if the boot version is lower than 2.4.5 ；
if user does not upgrade the boot version to 2.4.5 , his application may be not start.
i still suggest you be compatible this.

Reply:
The pr supported feature is submitted to Spring Cloud Alibaba 2021.x based on SpringBoot 2.4.x, so users won't exist in versions lower than SpringBoot 2.4.x?

Reply:
In fact, I have been testing, user does not have problems when the version is below 2.4.5, and the application can start and work properly.

The current SCA use boot 2.4.2, currently remove this feature, when SCA use boot 2.4.5 and above, then add this feature ?  I think this is a solution.

Reply:
Fixed